File: app/src/main/java/com/checkin/app/checkin/Account/AccountRepository.java
package com.checkin.app.checkin.Account;

import android.app.Application;
import android.content.Context;

import androidx.annotation.NonNull;
import androidx.lifecycle.LiveData;

import com.checkin.app.checkin.data.network.ApiClient;
import com.checkin.app.checkin.data.network.ApiResponse;
import com.checkin.app.checkin.data.BaseRepository;
import com.checkin.app.checkin.data.resource.NetworkBoundResource;
import com.checkin.app.checkin.data.resource.Resource;
import com.checkin.app.checkin.data.network.RetrofitLiveData;
import com.checkin.app.checkin.data.network.WebApiService;

import java.util.List;

public class AccountRepository extends BaseRepository {
    private static AccountRepository INSTANCE;
    private WebApiService mWebService;

    private AccountRepository(Context context) {
        mWebService = ApiClient.Companion.getApiService(context);
    }

    public static AccountRepository getInstance(Application application) {
        if (INSTANCE == null) {
            synchronized (AccountRepository.class) {
                if (INSTANCE == null) {
                    INSTANCE = new AccountRepository(application.getApplicationContext());
                }
            }
        }
        return INSTANCE;
    }

    public LiveData<Resource<List<AccountModel>>> getSelfAccounts() {
        return new NetworkBoundResource<List<AccountModel>, List<AccountModel>>() {
            @Override
            protected boolean shouldUseLocalDb() {
                return false;
            }

            @NonNull
            @Override
            protected LiveData<ApiResponse<List<AccountModel>>> createCall() {
                return new RetrofitLiveData<>(mWebService.getSelfAccounts());
            }

            @Override
            protected void saveCallResult(List<AccountModel> data) {

            }
        }.getAsLiveData();
    }
}

File: app/src/main/java/com/checkin/app/checkin/accounts/AccountRepository.kt
package com.checkin.app.checkin.accounts

import android.app.Application
import android.content.Context
import androidx.lifecycle.LiveData
import com.checkin.app.checkin.data.BaseRepository
import com.checkin.app.checkin.data.db.AppDatabase
import com.checkin.app.checkin.data.network.ApiClient.Companion.getApiService
import com.checkin.app.checkin.data.network.ApiResponse
import com.checkin.app.checkin.data.network.RetrofitLiveData
import com.checkin.app.checkin.data.network.WebApiService
import com.checkin.app.checkin.data.resource.NetworkBoundResource
import com.checkin.app.checkin.data.resource.Resource
import com.checkin.app.checkin.utility.SingletonHolder
import io.objectbox.android.ObjectBoxLiveData

class AccountRepository private constructor(context: Context) : BaseRepository() {
    private val mWebService: WebApiService = getApiService(context)
    private val boxAccount = AppDatabase.boxFor<AccountModel>()

    val selfAccounts: LiveData<Resource<List<AccountModel>>>
        get() = object : NetworkBoundResource<List<AccountModel>, List<AccountModel>>() {
            override fun shouldUseLocalDb(): Boolean = true

            override fun createCall(): LiveData<ApiResponse<List<AccountModel>>> {
                return RetrofitLiveData(mWebService.selfAccounts)
            }

            override fun loadFromDb(): LiveData<List<AccountModel>>? = ObjectBoxLiveData(boxAccount.query().build())

            override fun saveCallResult(data: List<AccountModel>?) {
                if (data != null) {
                    boxAccount.removeAll()
                    boxAccount.put(data)
                }
            }
        }.asLiveData

    companion object : SingletonHolder<AccountRepository, Application>({ AccountRepository(it.applicationContext) })
}